presto-hive-common/src/test/java/com/facebook/presto/hive/zorder/TestZOrderIntegers.java

```diff
@@ -0,0 +1,269 @@
+        }
+
+        ZOrder zOrder = new ZOrder(bitPositions);
+
+        try {
+            zOrder.encodeToByteArray(intColumns);
+            fail(format("Expected test to fail: z-ordering does not support more than %d integers.", ZOrder.MAX_INPUT_DIMENSIONS));
+        }
+        catch (IllegalArgumentException e) {
+            String expectedMessage = format("Current Z-Ordering implementation does not support more than %d input numbers.", ZOrder.MAX_INPUT_DIMENSIONS);
+            assertEquals(e.getMessage(), expectedMessage, format("Expected exception message '%s' to match '%s'", e.getMessage(), expectedMessage));
+        }
+    }
+
+    @Test
+    public void testZOrderNullInput()
+    {
+        ZOrder zOrder = new ZOrder(ImmutableList.of(8, 8));
+
+        try {
+            zOrder.encodeToByteArray(null);
+            fail("Expected test to fail: input list should not be null.");
+        }
+        catch (NullPointerException e) {
+            String expectedMessage = "Input list should not be null.";
+            assertEquals(e.getMessage(), expectedMessage, format("Expected exception message '%s' to match '%s'", e.getMessage(), expectedMessage));
+        }
+    }
+
+    @Test
+    public void testZOrderEmptyInput()
+    {
+        ZOrder zOrder = new ZOrder(ImmutableList.of(8, 8));
+        List<Integer> intColumns = ImmutableList.of();
+
+        try {
+            zOrder.encodeToByteArray(intColumns);
+            fail("Expected test to fail: input size should be greater than zero.");
+        }
+        catch (IllegalArgumentException e) {
+            String expectedMessage = "Input list size should be greater than zero.";
+            assertEquals(e.getMessage(), expectedMessage, format("Expected exception message '%s' to match '%s'", e.getMessage(), expectedMessage));
+        }
+    }
+
+    @Test
+    public void testZOrderNullEncodingBits()
+    {
+        try {
+            new ZOrder(null);
+            fail("Expected test to fail: encoding bits list should not be null.");
+        }
+        catch (NullPointerException e) {
+            String expectedMessage = "Encoding bits list should not be null.";
+            assertEquals(e.getMessage(), expectedMessage, format("Expected exception message '%s' to match '%s'", e.getMessage(), expectedMessage));
+        }
+    }
+
+    @Test
+    public void testZOrderEmptyEncodingBits()
+    {
+        try {
+            new ZOrder(ImmutableList.of());
+            fail("Expected test to fail: encoding bits list should not be empty.");
+        }
+        catch (IllegalArgumentException e) {
+            String expectedMessage = "Encoding bits list should not be empty.";
+            assertEquals(e.getMessage(), expectedMessage, format("Expected exception message '%s' to match '%s'", e.getMessage(), expectedMessage));
+        }
+    }
+
+    @Test
+    public void testZOrderToByteArray()
+    {
+        Random rand = new Random();
```

Comment:
Instead of using random, can we have some concrete numbers to validate the addresses are expected? For example, we can just take the example in https://en.wikipedia.org/wiki/Z-order_curve. The reason is to avoid any mistakes made by `encode` that could be corrected by "decode". So we also wanna validate the output is expected without a full roundtrip.